uaclient/api/u/pro/unattended_upgrades/status/v1.py

```diff
@@ -0,0 +1,188 @@
+    if not systemd_apt_timer_enabled:
+        return (False, messages.UNATTENDED_UPGRADES_SYSTEMD_JOB_DISABLED.msg)
+
+    for key, value in unattended_upgrades_cfg.items():
+        if not value:
+            return (
+                False,
+                messages.UNATTENDED_UPGRADES_CFG_LIST_VALUE_EMPTY.format(
+                    cfg_name=key
+                ).msg,
+            )
+        if isinstance(value, str) and value == "0":
+            return (
+                False,
+                messages.UNATTENDED_UPGRADES_CFG_VALUE_TURNED_OFF.format(
+                    cfg_name=key
+                ).msg,
+            )
+
+    return (True, "")
+
+
+def _get_unattended_upgrades_last_run() -> Optional[datetime.datetime]:
+    try:
+        creation_epoch = os.path.getctime(UNATTENDED_UPGRADES_STAMP_PATH)
+    except FileNotFoundError:
+        return None
+
+    return datetime.datetime.fromtimestamp(creation_epoch)
+
+
+def status() -> UnattendedUpgradesStatusResult:
+    return _status(UAConfig())
+
+
+def _status(cfg: UAConfig) -> UnattendedUpgradesStatusResult:
+    systemd_apt_timer_enabled = _get_apt_daily_job_status()
+    unattended_upgrades_last_run = _get_unattended_upgrades_last_run()
+
+    unattended_upgrades_cfg = get_apt_config_values(
+        set(
+            UNATTENDED_UPGRADES_CONFIG_KEYS
+            + get_apt_config_keys("Unattended-Upgrade")
+        )
+    )
+
+    # If that key is not present on the APT Config, we assume it
+    # that the config is "enabled", as by default this configuration
+    # will not be present in APT
+    unattended_upgrades_cfg["APT::Periodic::Enable"] = (
+        unattended_upgrades_cfg["APT::Periodic::Enable"] or "1"
+    )
+
+    (
+        unattended_upgrades_running,
+        disabled_reason,
+    ) = _is_unattended_upgrades_running(
+        systemd_apt_timer_enabled, unattended_upgrades_cfg
+    )
+
+    unattended_upgrades_result = UnattendedUpgradesStatusResult(
+        systemd_apt_timer_enabled=systemd_apt_timer_enabled,
+        apt_periodic_job_enabled=str(
+            unattended_upgrades_cfg.get("APT::Periodic::Enable", "")
+        )
+        == "1",
+        package_lists_refresh_frequency_days=int(
+            unattended_upgrades_cfg.get(  # type: ignore
+                "APT::Periodic::Update-Package-Lists", 0
+            )
+        ),
+        unattended_upgrades_frequency_days=int(
+            unattended_upgrades_cfg.get(  # type: ignore
+                "APT::Periodic::Unattended-Upgrade", 0
+            )
+        ),
+        unattended_upgrades_allowed_origins=list(
+            unattended_upgrades_cfg.get("Unattended-Upgrade::Allowed-Origins")
+            or []
+        ),
+        unattended_upgrades_running=unattended_upgrades_running,
+        unattended_upgrades_disabled_reason=disabled_reason,
+        unattended_upgrades_last_run=unattended_upgrades_last_run,
+    )
+    unattended_upgrades_result.meta = unattended_upgrades_cfg
```

Comment:
my earlier suggestion probably led you astray, but in the spec, we said this would be under the `raw_config` key inside meta.